pkg/apis/extensions/types.go

```diff
@@ -899,3 +899,45 @@ type NetworkPolicyList struct {
 
 	Items []NetworkPolicy `json:"items"`
 }
+
+// +genclient=true
+// +nonNamespaced=true
+
+// StorageClass describes the parameters for a class of storage for
+// which PersistentVolumes can be dynamically provisioned.
+//
+// StorageClasses are non-namespaced; the name of the storage class
+// according to etcd is in ObjectMeta.Name.
+type StorageClass struct {
+	unversioned.TypeMeta `json:",inline"`
+	api.ObjectMeta       `json:"metadata,omitempty"`
+
+	// ProvisionerType indicates the type of the provisioner.
+	ProvisionerType string `json:"provisionerType,omitempty"`
```

Comment:
Is this intended to be optional?  If not, remove the `omitempty` and add details to the comments that it may not be empty.  If so, make it a pointer and detail in comments what happens if it is not specified.